Feature/verify creds

Siunami opened the pull request.

Siunami committed:
verify component flow

Siunami committed:
verifying attendee credentials

Siunami committed:
fix merge conflicts

Siunami committed:
removing console statements

rmw2 commented on src/components/events/verify/Verify.js:
```diff
@@ -0,0 +1,111 @@
+ *
+ */
+// const VerifyEvent = ({did}) => {
+class VerifyEvent extends Component {
```
Can we call this `AttendanceVerifier` or `EventVerifier` or something that fits with `EventCreator` and `EventCheckinAttestor`

rmw2 commented on src/components/events/verify/Verify.js:
```diff
@@ -0,0 +1,111 @@
+        this.createEventIdentity(this.props)
+    }
+
+    returnDashboard = (event) => {
```
Let's keep instance methods all in the same style and not mix arrow functions with class methods.  This one may actually be able to just be inlined on line 83, i.e. `onClick={() => browerHistory.push('/dashboard')}`

rmw2 commented on src/components/events/verify/Verify.js:
```diff
@@ -0,0 +1,111 @@
+        this.waitForCheckin()
+      }
+
+    doCheckin (data){
```
Let's change this to `doVerify` or something that doesn't confuse it with the checkin page

rmw2 commented on src/components/events/verify/Verify.js:
```diff
@@ -0,0 +1,111 @@
+        const verified = data.verified
+        let notif = document.getElementById('verified')
+        for (var i = 0; i < verified.length; i++){
+            if (verified[i].claim.uportLiveEvent.identifier.did == this.state.event &&
```
So this is where we need two different methods for verifying, depending on which type of event attestation they received.  In the case of an ethr-did signed attendance attestation, we just need `verified[i].iss === event`.  For the credentials with an event did's signature injected in the claim body, we should do something like
```javascript
import {verifyJWT} from 'did-jwt'
// ...
const { signature } = verified[i].claim.uportLiveEvent
verifyJWT(signature).then(({ issuer, payload }) => {
  if (issuer === event && payload.attendee === data.address) {
    // Verified successfully
  }
})
```

This whole thing should probably be extracted into a separate function to save clutter and make it reusable

rmw2 commented on src/components/events/verify/Verify.js:
```diff
@@ -0,0 +1,111 @@
+            }
+            // console.log(verified[i].claim.uportLiveEvent.identifier.did)
+        }
+        notif.innerHTML = "Sorry not an attendee <i class='exclamation icon'></i>"
```
Can we make a more elegant react version of the success/fail dialog?  Maybe add an `isVerified` field to `this.state` and then conditionally display text based on that.  Might even be nice to do it in a pop-up modal or something, but generally we shouldn't try to mix direct dom manipulation with react, both for style reasons and bc React doesn't always play nice with dom hacking

rmw2 commented on src/components/events/dashboard/EventCard.js:
```diff
@@ -18,6 +19,11 @@ const EventCard = ({beginCheckin, isActive, ...eventData}) => {
 
+  const verify = (event) => {
+    event.preventDefault()
+    browserHistory.push('/verify/?event=' + eventData.identifier.did)
```
Trailing slash isn't necessary and could goof things `/verify/?event=` => `/verify?event=`

Also since this onClick isn't on a form submission or another button with predefined behavior, we don't need to `preventDefault()`

rmw2 commented:
Re: notifying verification success/failure in a modal, branch [feature/general-modal](https://github.com/uport-project/uPort-live/tree/feature/general-modal) has support for popping up arbitrary modals, so you can put that part off til the modal branch gets merged in and use that

Siunami committed:
verifyJWT error

rmw2 committed:
WIP skeleton for two-types of verification, fixed missing verify component

rmw2 committed:
Merging in modal support from master

rmw2 committed:
Updated modal component registration

Export functions registerModal and whichModal from misc/modal/actions, so that new modals can be registered from anywhere and don't require editing a central modals file. Still could be a little bit cleaner, but this is making progress.

rmw2 committed:
Fixed verification flow and now use modals for success/failure

Added new images failure.png and success.png for use in the verify page
Fixed the checkin attestor to use the ES256K-R algorithm instead of ES256K to get rid of the '-1 is not property of null' bug.  Also added issue uport-project/did-jwt#14 documenting the wackness.
Made sure that the modal doesn't appear until after all verifications are tested.

rmw2 commented:
TODO:
- [ ] Provide an interface to enter an event did from the `/verify` page if none is provided in the query parameters
- [ ] Improve style/text on the verify page
- [ ] Add verify link to old events on dashboard
- [ ] Link to verify page from homepage
- [ ] Add UI to easily copy the link to a verifier page
- [ ] Allow the `/verify` page to do a verification on multiple events
